README.md

```diff
@@ -103,6 +107,23 @@ def test_during_day_light_DOES_NOT_turn_on(given_that, living_room, assert_that)
     assert_that('light.living_room').was_not.turned_on()
 ```
 
+### Deprecation Warnings
+As development continues of this test framework, some interfaces and test fixtures need to get
+deprecated. In general, the following method is used to ease the transitions
+1. Mark deprecated calls with a warning
+1. Provide an expressive message directing you how to change your code
+1. At least one release will include the warning
```

Comment:
I'm thinking more in terms of months. We could say "At least 6 months (1 year?)" for instance. People setting up automation systems, once it's working, are often in 'set and forget' mode. So a loooong deprecation period is important I believe.

But I'm happy to discuss about it if you feel it would clutter the code for too long.

Reply:
Another option rather than setting a specific time on it would be to follow semantic versioning and say that any deprecated function will not be removed until the next major release. That way if people are properly specing their dependencies (e.g. `appdaemontestframework >=2.6.1,<3.0`) they would not break upon release of a new major version, they would need to manually go out an update and review breaking changes in the new major version on that point.

Also, either way it might make sense to keep a changelog to help communicate these. I'm partial to the keep a change log format: https://keepachangelog.com/en/1.0.0/

Reply:
Yep. Using major releases for breaking changes makes sense. Keeping a changelog is a great idea.
So far I documented all changes as messages of the `Release` commits. So the changelog is visible there: https://github.com/FlorianKempenich/Appdaemon-Test-Framework/releases?after=v2.2.0
But I agree an official changelog would make sense. I'm going to create an issue for it.

Reply:
https://github.com/FlorianKempenich/Appdaemon-Test-Framework/issues/29